tapir/coop/views.py

```diff
@@ -490,6 +505,113 @@ class CurrentShareOwnerListView(
     ordering = ["-share_ownerships__start_date"]
 
 
+class ShareOwnerTable(django_tables2.Table):
+    display_name = django_tables2.Column(
+        empty_values=(), verbose_name="Name", orderable=False
+    )
+    status = django_tables2.Column(empty_values=(), orderable=False)
+    email = django_tables2.Column(empty_values=(), orderable=False, visible=False)
+    ratenzahlung = django_tables2.Column(
+        empty_values=(), orderable=False, visible=False
+    )
+    phone_number = django_tables2.Column(
+        empty_values=(), orderable=False, visible=False
+    )
+
+    def render_display_name(self, value, record: ShareOwner):
+        return format_html(
+            "<a href={}>{}</a>",
+            record.get_absolute_url(),
+            record.get_info().get_display_name(),
+        )
+
+    def value_display_name(self, value, record: ShareOwner):
+        return record.get_info().get_display_name()
+
+    def render_status(self, value, record: ShareOwner):
+        status = record.get_member_status()
+        if status == ShareOwner.MEMBER_STATUS_SOLD:
+            color = "orange"
+        elif status == ShareOwner.MEMBER_STATUS_ACTIVE:
+            color = "green"
+        else:
+            color = "blue"
+
+        return format_html('<span style="color: {1};">{0}</span>', status, color)
+
+    def value_status(self, value, record: ShareOwner):
+        return record.get_member_status()
+
+    def value_email(self, value, record: ShareOwner):
+        return record.get_info().email
+
+    def value_phone_number(self, value, record: ShareOwner):
+        return record.get_info().phone_number
+
+    class Meta:
+        model = ShareOwner
+        template_name = "django_tables2/bootstrap4.html"
+        fields = ["id", "from_startnext"]
+        sequence = ("id", "display_name", "status", "from_startnext")
+
+
+class ShareOwnerFilter(django_filters.FilterSet):
+    display_name = CharFilter(method="display_name_filter", label="Name")
+    status = ChoiceFilter(
+        choices=(
+            (ShareOwner.MEMBER_STATUS_ACTIVE, ShareOwner.MEMBER_STATUS_ACTIVE),
```

Comment:
See comment in `models.py`, we should separate out the constants from the user-visible translatable strings,